napari/utils/stubgen.py

```diff
@@ -86,6 +86,9 @@ def generate_function_stub(func) -> Tuple[Set[str], str]:
     """Generate a stub and imports for a function."""
     sig = inspect.signature(func)
 
+    if hasattr(func, "__wrapped__"):
```

Comment:
I guess this is fine, but why do we need to unwrap? I.e. what is it about the wrapped function that causes an error? Shouldn't the important function attributes be passed through to the wrapped function?

Is this because we add `_forwardrefns_` in `_merge_layer_viewer_sigs_docs`, but this is ignored by `wraps`? If so, I guess I can't think of a better solution.

Reply:
because line below there is the construction of `globalns` variable in the context of a function, so in this case in the context of the wrapper, not the wrapped function. 

This is the same what `get_type_hints` perform if `globals` is not provided.